org/mozilla/jss/pkcs11/attrs/CKAUsage.java

```diff
@@ -0,0 +1,115 @@
+package org.mozilla.jss.pkcs11.attrs;
+
+import org.mozilla.jss.pkcs11.PKCS11Constants;
+
+/**
+ * CKAUsage is a collection of PKCS#11 CK_ATTRIBUTES which have common value
+ * (CK_TRUE).
+ */
+public class CKAUsage extends CKAttribute {
+    public CKAUsage() {}
```

Comment:
This constructor allows `CKAUsage` to be created with uninitialized/default `type`. If this is not allowed this constructor probably should be removed.

Reply:
Ah good catch. I forgot to add the `setType(...)` helper for this. There's a few NSS extensions to PKCS#11 that could be set here, that I haven't added classes for. So `CKAUsage() + setType()` should be possible. I don't want to add `setType(...)` to CKAttribute as it really shouldn't be necessary to set the type on other attributes.

Reply:
Actually, I take that back. The one argument constructor is probably best here. I'll drop the `setType(...)` and no-argument constructor.

Reply:
Or we could just add a default value, if applicable.

Reply:
Most of the time there isn't a default value. Only `CKA_VALUE_LENGTH` really has a default value.